R/variables.R

```diff
@@ -212,13 +215,13 @@ predict.GradientBoost <- function(object, newdata = NULL, ...)
                  "If errors persist after recomputing, please contact support for further help")
     }
 
-    if (object$original$params$objective == "binary:logistic")
+    if (object$original$params$objective == "binary:logistic" && !keep.soft.probs)
     {
         prediction <- ceiling(2 * prediction)
         prediction <- object$outcome.levels[prediction]
         prediction <- factor(prediction, levels = object$outcome.levels)
     }
-    if (object$original$params$objective == "multi:softprob")
+    if (object$original$params$objective == "multi:softprob" && !keep.soft.probs)
     {
         prediction <- factor(apply(prediction, 1, which.max), levels = as.character(1:length(object$outcome.levels)))
         levels(prediction) <- object$outcome.levels
```

Comment:
assign correct levels in the `factor` call

Reply:
Updated in https://github.com/Displayr/flipMultivariates/pull/8/commits/5e7a2992ec8ed0598f29f19e7e3b54470916fd86